app/models/reception/resource_factory.rb

```diff
@@ -148,5 +144,37 @@ def duplicate_containers
       cons.fetch('Well', []).map { |well| [well.position, well.plate] }.uniq! ||
         cons.fetch('Tube', []).map(&:barcode).uniq! || cons.fetch('Plate', []).map(&:barcode).uniq!
     end
+
+    def library_type_cache
+      # Used to reduce database queries.
+      @library_type_cache ||= LibraryType.all.index_by(&:name)
+    end
+  end
+
+  # A helper that deals with the creation of library type specific records.
+  class LibraryTypeHelper
+    def initialize(library_type, resource_factory, container_attributes)
+      @library_type = library_type
+      @resource_factory = resource_factory
+      @container_attributes = container_attributes
+    end
+
+    def create_library(request)
+      # The library type is used to help build the correct library in the correct pipeline
+      return unless @container_attributes[:library]
```

Comment:
Would this return if we send an empty library object?

Reply:
No, it tries to create the library.  I'm writing the test for it doing that.  I think if someone's providing a `library` key they probably meant to create a library and not doing so silently would be confusing.